### PR TITLE
SALTO-2041 fix netsuite references in workato

### PR DESCRIPTION
**Pull request by dantal4**
expect `customrecordtype.customrecordcustomfields.customrecordcustomfield` to be map instead of list 

---
_Release Notes_: 
- fix bug - iterate map values instead of lists items 

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/workato-adapter/src/filters/cross_service/netsuite/element_index.ts [hunk 1]
```diff
@@ -47,13 +49,14 @@ export const indexNetsuiteByTypeAndScriptId = (
...
-            .map((f: { scriptid: string }, idx: number) => ({
+          Object.values(inst.value[CUSTOM_RECORD_CUSTOM_FIELDS]?.[CUSTOM_RECORD_CUSTOM_FIELD] ?? {})
+            .filter(isPlainRecord)
+            .map(f => ({
               scriptId: f.scriptid,
               nestedPath: inst.elemID.createNestedID(
                 CUSTOM_RECORD_CUSTOM_FIELDS,
                 CUSTOM_RECORD_CUSTOM_FIELD,
-                String(idx),
+                String(f.index),
```
Comment (netama) [inline packages/workato-adapter/src/filters/cross_service/netsuite/element_index.ts:59]: I think this should be the map's key, no? it is creating an elem id

File: packages/workato-adapter/test/filters/recipe_references.test.ts [hunk 2]
```diff
@@ -862,11 +862,11 @@ describe('Recipe references filter', () => {
         scriptid: 'customrecord16',
         recordname: 'my custom record',
         customrecordcustomfields: {
-          customrecordcustomfield: [
-            { scriptid: 'custrecord5' },
-            { scriptid: 'somethingelse' },
-            { scriptid: 'custrecordaccount_id' },
-          ],
+          customrecordcustomfield: {
+            custrecord5: { scriptid: 'custrecord5', index: 0 },
```
Comment (netama) [inline packages/workato-adapter/test/filters/recipe_references.test.ts:866]: the referencse in https://github.com/salto-io/salto/blob/main/packages/workato-adapter/test/filters/recipe_references.test.ts#L1020 should also change (and be valid elem ids)

File: packages/workato-adapter/src/filters/cross_service/netsuite/element_index.ts (hunk 1 above)
Comment (dantal4) [reply packages/workato-adapter/src/filters/cross_service/netsuite/element_index.ts:59]: you're right. fixing it

File: packages/workato-adapter/src/filters/cross_service/netsuite/element_index.ts [hunk 3]
```diff
@@ -47,15 +49,21 @@ export const indexNetsuiteByTypeAndScriptId = (
...
       customRecordTypeInstances
         .flatMap(inst => (
-          (inst.value[CUSTOM_RECORD_CUSTOM_FIELDS]?.[CUSTOM_RECORD_CUSTOM_FIELD] ?? [])
-            .map((f: { scriptid: string }, idx: number) => ({
-              scriptId: f.scriptid,
+          Object.entries(
+            inst.value[CUSTOM_RECORD_CUSTOM_FIELDS]?.[CUSTOM_RECORD_CUSTOM_FIELD] ?? {}
+          ).map(([key, item]) => {
+            if (!isPlainRecord(item)) {
+              return undefined
```
Comment (netama) [inline packages/workato-adapter/src/filters/cross_service/netsuite/element_index.ts:56]: is this a valid scenario? can we have a log?

Comment (dantal4) [reply packages/workato-adapter/src/filters/cross_service/netsuite/element_index.ts:56]: shouldn't be. it's a type guard only to allow accessing `item.scriptid` later.

**Pull request merged**